### PR TITLE
Refactor Admission Webhook templates and values

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Current Admission Webhook templates is complex (which means hard to maintain) and the values setting is misleading and difficult to understand.

This request is to simplify the admission templates and make admission values setting easier to understand.

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
ACTION REQUIRED:
1. Change the setting from the previous admission.hookEnabled.pods to the admission.validation.pods.
2. Change the setting from the previous admission.hookEnabled.statefulSets to the admission.validation.statefulSets
3. Change the setting from the previous admission.hookEnabled.validating to the admission.validation.pingcapResources.
4. Change the setting from the previous admission.hookEnabled.defaulting to the admission.mutation.pingcapResources.
5. Change the setting from the previous admission.failurePolicy.defaulting to the admission.failurePolicy.mutation.
6. Change the setting from the previous admission.failurePolicy.OTHER SETTING to the
admission.failurePolicy.validation.
```

### CODE REVIEW

File: charts/tidb-operator/values.yaml [hunk 1]
```diff
@@ -185,30 +185,27 @@ admissionWebhook:
...
-    ## updateStatefulSet Webhook would check the updating request of tidbcluster statefulset and the failurePolicy is recommended as Ignore
-    updateStatefulSet: Ignore
-    ## validation hook validates the correctness of the resources under pingcap.com group
-    validating: Ignore
-    ## defaulting hook set default values for the the resources under pingcap.com group
-    defaulting: Ignore
+    ## the validation webhook would check the request of the given resources, and the failurePolicy is recommended as Fail
+    validation: Fail
+    ## the mutation webhook would mutate th erequest of the given resources, and the failurePolicy is recommended as Ignore
+    mutation: Ignore
```
Comment (Yisaer) [inline charts/tidb-operator/values.yaml]: For the failurePlicy, I tend to use `validation` and `mutation` to cover all the admission webhooks in the coarse-grained level

File: charts/tidb-operator/values.yaml [hunk 2]
```diff
@@ -185,30 +185,27 @@ admissionWebhook:
   ## if admissionWebhook.create and admissionWebhook.hooksEnabled.pods are both enabled,
   ## The pre-delete-job would delete the validationWebhookConfiguration using this image
   jobImage: "bitnami/kubectl:latest"
-  hooksEnabled:
```
Comment (Yisaer) [inline charts/tidb-operator/values.yaml]: hooksEnabled is confusing, remove it and use `validation` and `mutation` instead.

File: charts/tidb-operator/templates/admission/admission-webhook-registration.yaml [hunk 3]
```diff
@@ -130,11 +101,11 @@ webhooks:
         resources: ["statefulsets"]
 {{- end }}
 ---
-{{- if .Values.admissionWebhook.hooksEnabled.validating }}
+{{- if .Values.admissionWebhook.validation.pingcapResources }}
 apiVersion: admissionregistration.k8s.io/v1beta1
 kind: ValidatingWebhookConfiguration
 metadata:
-  name: pingcap-resources-validating
+  name: pingcap-tidb-resources-validating
```
Comment (Yisaer) [inline charts/tidb-operator/templates/admission/admission-webhook-registration.yaml]: As ChaosMesh released, tidbcluster is not the only resources under pingcap group.

File: charts/tidb-operator/values.yaml [hunk 4]
```diff
@@ -185,30 +185,27 @@ admissionWebhook:
...
+  validation:
     ## statefulsets hook would check requests for updating tidbcluster's statefulsets
     ## If enabled it, the statefulsets of tidbcluseter would update in partition by tidbcluster's annotation
     statefulSets: false
     ## pods hook would check requests for creating and deleting tidbcluster's pods
     ## if enabled it, the pods of tidbcluster would safely created or deleted by webhook instead of controller
     pods: true
     ## validating hook validates the correctness of the resources under pingcap.com group
-    validating: false
+    pingcapResources: false
```
Comment (DanielZhangQD) [inline charts/tidb-operator/values.yaml]: Do we support `true` now? If yes, why set it to `false` by default?

Comment (Yisaer) [reply charts/tidb-operator/values.yaml]: I guess the resource validating is not verified yet.

File: charts/tidb-operator/values.yaml [hunk 5]
```diff
@@ -185,30 +185,27 @@ admissionWebhook:
...
-  hooksEnabled:
+  ## validation webhook would check the given request for the specific resource and operation
+  validation:
     ## statefulsets hook would check requests for updating tidbcluster's statefulsets
     ## If enabled it, the statefulsets of tidbcluseter would update in partition by tidbcluster's annotation
     statefulSets: false
     ## pods hook would check requests for creating and deleting tidbcluster's pods
     ## if enabled it, the pods of tidbcluster would safely created or deleted by webhook instead of controller
     pods: true
     ## validating hook validates the correctness of the resources under pingcap.com group
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply charts/tidb-operator/values.yaml]: Any advice?

**Comment by Yisaer** [comment]
The failurePolicy of validation is changed to `Ignore` now @DanielZhangQD PTAL

File: charts/tidb-operator/values.yaml (hunk 1 above)
Comment (Yisaer) [reply charts/tidb-operator/values.yaml]: updated @DanielZhangQD

File: charts/tidb-operator/values.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply charts/tidb-operator/values.yaml]: I mean you change the field name but the name in the description is not changed.

File: charts/tidb-operator/values.yaml [hunk 6]
```diff
@@ -185,30 +185,27 @@ admissionWebhook:
...
-    ## createPod Webhook would check the creating request of tidbcluster pod and the failurePolicy is recommended as Ignore
-    createPod: Ignore
-    ## updateStatefulSet Webhook would check the updating request of tidbcluster statefulset and the failurePolicy is recommended as Ignore
-    updateStatefulSet: Ignore
-    ## validation hook validates the correctness of the resources under pingcap.com group
-    validating: Ignore
-    ## defaulting hook set default values for the the resources under pingcap.com group
-    defaulting: Ignore
+    ## the validation webhook would check the request of the given resources, and the failurePolicy is recommended as Ignore
+    validation: Ignore
```
Comment (DanielZhangQD) [inline charts/tidb-operator/values.yaml]: So `ignore` for deleting Pod?
If createPod and deletePod do need different failurePolicy, why do we combine all the fields to only one?

File: charts/tidb-operator/values.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Yisaer) [reply charts/tidb-operator/values.yaml]: good catch, updated.

File: charts/tidb-operator/values.yaml (hunk 6 above)
Comment (Yisaer) [reply charts/tidb-operator/values.yaml]: They are both recommended being setting as `Ignore`, but they still need to be managed separetely.

Comment (DanielZhangQD) [reply charts/tidb-operator/values.yaml]: But they are both controlled by `validation` now, you can not set different failurePolicy for createPod and deletePod, right?

**Comment by Yisaer** [comment]
Thanks all. This really helps refine the values settings.

**Pull request merged**